processing/src/main/java/io/druid/query/search/search/RegexSearchQuerySpec.java

```diff
@@ -0,0 +1,108 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.query.search.search;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+import java.util.regex.Pattern;
+
+
+public class RegexSearchQuerySpec implements SearchQuerySpec
+{
+
+  private static final byte CACHE_TYPE_ID = 0x3;
+
+  private final String pattern;
+  private final Pattern compiled;
+
+  @JsonCreator
+  public RegexSearchQuerySpec(
+      @JsonProperty("pattern") String pattern
+  )
+  {
+    this.pattern = pattern;
+    compiled = Pattern.compile(pattern);
```

Comment:
preconditions notnull with a hint to `pattern` is the typical workflow for things such as this. It helps debug on querying side.